R/get_meetnetten_locations.R

```diff
@@ -0,0 +1,178 @@
+#' sublocations <- locations_dragonflies$sublocations
+#'
+#' # Close the connection when done
+#' dbDisconnect(con)
+#' rm(con)
+#' }
+
+get_meetnetten_locations <- function(connection,
+                                    scheme_name = NULL,
+                                    species_group = NULL) {
+
+  species_group_selected <- species_group
+
+  assert_that(inherits(connection, what = "Microsoft SQL Server"),
+              msg = "Not a connection object to database.")
+
+  if (!is.null(scheme_name)) {
+    assert_that(is.character(scheme_name))
+    scheme_name <- str_to_lower(scheme_name)
+  }
+
+  if (!is.null(species_group_selected)) {
+    assert_that(is.character(species_group_selected))
+    species_group_selected <- str_to_lower(species_group_selected)
+  }
+
+  sql_statement <- "SELECT
+    PG.name as species_group,
+    P.name as scheme,
+    L.name as location,
+    L.id,
+    L.parent_id,
+    PL.is_sample,
+    PL.is_active,
+    L.geom.STAsText() as geom
+    FROM staging_meetnetten.locations_location L
+    INNER JOIN staging_meetnetten.projects_projectlocation PL ON PL.location_id = L.id
+    INNER JOIN staging_meetnetten.projects_project P ON P.id = PL.project_id
+    INNER JOIN staging_meetnetten.projects_projectgroup PG on PG.id = P.group_id"
+
+  query_result <- tbl(connection, sql(sql_statement))
+
+  if (!is.null(scheme_name) & !is.null(species_group_selected)) {
+
+    query_result <- query_result %>%
+      filter(str_to_lower(.data$scheme) %in% scheme_name |
+               str_to_lower(.data$species_group) %in% species_group_selected)
+
+  } else if (!is.null(scheme_name)) {
+
+    query_result <- query_result %>%
+      filter(str_to_lower(.data$scheme) %in% scheme_name)
+
+  } else if (!is.null(species_group_selected)) {
+
+    query_result <- query_result %>%
+      filter(str_to_lower(.data$species_group) %in% species_group_selected)
+
+  }
+
+  locations_sf <- query_result %>%
+    collect() %>%
+    mutate(location_type = ifelse(.data$parent_id == -1,
+                                  "main_location",
+                                  "sublocation")) %>%
+    st_as_sf(wkt = "geom", crs = 4326)
+
+  locations_main <- locations_sf %>%
+    filter(.data$location_type == "main_location") %>%
+    arrange(.data$species_group, .data$scheme, .data$location)
+
+  locations_main_names <- locations_main %>%
+    st_drop_geometry() %>%
+    select(scheme, location, location_id = id)
+
+  locations_sublocations <- locations_sf %>%
+    filter(.data$location_type == "sublocation") %>%
+    rename(sublocation = location, location_id = parent_id) %>%
+    left_join(locations_main_names, by = c("scheme", "location_id")) %>%
+    arrange(.data$species_group, .data$scheme, .data$location,
+            .data$sublocation) %>%
+    filter(!is.na(.data$location))
+
+  locations_main <- locations_main %>%
+    select(species_group, scheme, location, is_sample, is_active)
```

Comment:
```suggestion
    select("species_group", "scheme", "location", "is_sample", "is_active")
```
en check ook andere opmerkingen die `lintr::lint_package()` geeft